src/oc/hoc_oop.cpp

```diff
@@ -1263,6 +1277,20 @@ void hoc_object_component() {
                 if (!ISARRAY(sym) || sym->arayinfo->nsub != nindex) {
                     hoc_execerror(sym->name, ":not right number of subscripts");
                 }
+                if (narg) {
+                    // there are a few modeldb examples that use (index) instead
+                    // of [index] syntax for an array in this context. So we
+                    // have decided to keep allowing this legacy syntax for one
+                    // dimensional arrays.
+                    if (narg == 1) {
+                        hoc_push_ndim(1);
+                    } else {
+                        hoc_execerr_ext("%s.%s is array not function. Use %s[...] syntax",
```

Comment:
Codecov says this isn't covered; can we add a test for it?

Reply:
I'll do that.

Reply:
There seems to be a conflict between my codecov and yours.  The below coverage is a consequence of the test commit 
3d8a580

![image](https://user-images.githubusercontent.com/4395729/223517134-889b6a04-f914-4105-980c-1336ca0e538f.png)

Reply:
<img width="768" alt="Screenshot 2023-03-07 at 19 36 30" src="https://user-images.githubusercontent.com/6459623/223518000-a0e10f15-ff6b-4732-b66d-f549d26ce552.png">

I was just looking at the annotations in the GitHub diff view; I didn't run anything myself. Not sure why there would be a difference.

Reply:
Ok. I see. It is 1288 that is not executed. Actually, I believe that line is not accessible. But I'll look a bit more carefully.

Reply:
OK, if you think this is the best we can do then that's fine by me.